src/structures/GuildChannel.js

```diff
@@ -35,8 +35,9 @@ class GuildChannel extends Channel {
      * The type of the Guild Channel
      * @type {String}
      */
-    //Returns either text, voice, or the id of the channel
-    this.type = data.type == 0 ? 'text' : data.type == 2 ? 'voice' : data.type.toString();
+    if(data.type === 0) this.type = 'text';
```

Comment:
Can you be more specific about which part of the code (execution/concept) is horrific to you and what result you would prefer?